server/SchemaLoader.js

```diff
@@ -0,0 +1,127 @@
+"use strict";
+
+const EventEmitter = require("events");
+const { remote } = require("electron");
+const jsonfile = require("jsonfile");
+const mkdirp = require("mkdirp");
+const path = require("path");
+const fs = require("fs");
+
+const ValidationSchema = require("./model/ValidationSchema");
+
+class SchemaLoader extends EventEmitter {
+    constructor() {
+        super();
+
+        this.on("schemas-synced", function() {
+            this.loadLocalSchemas();
+        });
+    }
+
+    syncSchemas() {
+        let repo = remote.getGlobal("options").repo;
+
+        if (!!repo && !!repo.url && !!repo.apikey && !!repo.vendors && repo.vendors.length > 0) {
+            let endpoint = "/api/schemas/" + repo.vendors.join("%2C");
+
+            const req = new XMLHttpRequest();
+            req.open("GET", repo.url + endpoint);
+            req.setRequestHeader("apikey", repo.apikey);
+            req.send();
+
+            req.onreadystatechange = function() {
+                if (req.readyState === 4 && req.status === 200) {
```

Comment:
- Instead of `4`, can we use `XMLHttpRequest.DONE` or something like that?
- Shouldn't we handle errors as well?

Reply:
1) Historically terrible for cross-browser support, using a recent version of Chromium we can get away with it here.

2) Yep, changed it.